### PR TITLE
Java side for PlayServer codegen

### PR DESCRIPTION
**Pull request by ktoso**
Java part of https://github.com/akka/akka-grpc/issues/209

### CODE REVIEW

**Comment by ktoso** [comment]
Amended, should pass

File: play-interop-test-java/src/test/scala/akka/grpc/gen/PlayJavaRouterSpec.scala [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  implicit val sys = ActorSystem()
+  implicit val mat = ActorMaterializer()
+  implicit val ec = sys.dispatcher
+  implicit val patience = PatienceConfig(timeout = 3.seconds, interval = 15.milliseconds)
+
+  // serializers so we can test the requests
+  implicit val HelloRequestSerializer = example.myapp.helloworld.grpc.GreeterService.Serializers.HelloRequestSerializer
+  implicit val HelloReplySerializer = example.myapp.helloworld.grpc.GreeterService.Serializers.HelloReplySerializer
+
+  implicit def unmarshaller[T](implicit serializer: ProtobufSerializer[T], mat: Materializer): FromRequestUnmarshaller[T] =
```
Comment (raboof) [inline play-interop-test-java/src/test/scala/akka/grpc/gen/PlayJavaRouterSpec.scala]: Nice, I guess these could go into a utility class somewhere but OK :+1:

File: play-interop-test-java/src/test/scala/akka/grpc/gen/PlayJavaRouterSpec.scala [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+  implicit def toSourceUnmarshaller[T](implicit serializer: ProtobufSerializer[T], mat: Materializer): FromRequestUnmarshaller[Source[T, NotUsed]] =
+    Unmarshaller((ec: ExecutionContext) ⇒ (req: HttpRequest) ⇒ GrpcMarshalling.unmarshalStream(req)(serializer, mat))
+
+  implicit def marshaller[T](implicit serializer: ProtobufSerializer[T], mat: Materializer, codec: Codec): ToResponseMarshaller[T] =
+    Marshaller.opaque((response: T) ⇒ GrpcMarshalling.marshal(response)(serializer, mat, codec))
+
+  implicit def fromSourceMarshaller[T](implicit serializer: ProtobufSerializer[T], mat: Materializer, codec: Codec): ToResponseMarshaller[Source[T, NotUsed]] =
+    Marshaller.opaque((response: Source[T, NotUsed]) ⇒ GrpcMarshalling.marshalStream(response)(serializer, mat, codec))
+  // embedding rather than generating
```
Comment (raboof) [inline play-interop-test-java/src/test/scala/akka/grpc/gen/PlayJavaRouterSpec.scala]: ?

File: sbt-plugin/src/main/scala/akka/grpc/sbt/test/PlayScalaCompositeCodeGenerator.scala [hunk 3]
```diff
@@ -5,17 +5,19 @@
...
 
 /**
  * An easy to use accessor to be used reflectively in the main sbt project,
  * only used in the ReflectiveCodeGen plugin in the main sbt project, which is needed for the interop-tests subproject
  */
-class PlayCompositeCodeGenerator {
+class PlayScalaCompositeCodeGenerator {
   def instance(): protocbridge.ProtocCodeGenerator =
     new ProtocBridgeSbtPluginCodeGenerator(new CompositeCodeGenerator(Seq(
+      // scala
```
Comment (raboof) [inline sbt-plugin/src/main/scala/akka/grpc/sbt/test/PlayScalaCompositeCodeGenerator.scala]: Maybe not needed ;)

File: codegen/src/main/twirl/templates/JavaServer/Handler.scala.txt [hunk 4]
```diff
@@ -24,21 +24,47 @@ import akka.grpc.javadsl.GrpcExceptionHandler;
...
 
+  /**
+   * Creates a `HttpRequest` to `HttpResponse` handler that can be used in for example `Http().bindAndHandleAsync`
+   * for the generated partial function handler and ends with `StatusCodes.NotFound` if the request is not matching.
+   *
+   * Use `akka.grpc.scaladsl.ServiceHandler.concatOrNotFound` with `@{service.name}Handler.partial` when combining
+   * several services.
+   */
+  public static Function<HttpRequest, CompletionStage<HttpResponse>> create(@service.name implementation, Materializer mat) {
+    return create(implementation, "/", mat);
```
[the remark this answers is not in the document]
Comment (ktoso) [reply codegen/src/main/twirl/templates/JavaServer/Handler.scala.txt]: you're right thanks

**Comment by ktoso** [comment]
Whoops `[error] java.lang.RuntimeException: There are files without headers!`

**Pull request merged**